### PR TITLE
Changed main callback return values to an enumeration

### PR DESCRIPTION
**Pull request by slouken**
Fixes https://github.com/libsdl-org/SDL/issues/10515

### CODE REVIEW

**Review by icculus — APPROVED** [review]
Looks good.

I probably need to have wikiheaders make redirect pages for individual enum members, but that's a different task.

**Comment by icculus** [comment]
> I probably need to have wikiheaders make redirect pages for individual enum members, but that's a different task.

Did this now, in abdd8b49291e48ccd3e495c54e95dccf8ebc99b1, So now when someone clicks on SDL_APP_CONTINUE (etc) in the example apps' source code tab, it goes to a real wiki page instead of a "never heard of that before" page.  :)

File: include/SDL3/SDL_init.h [hunk 1]
```diff
@@ -66,6 +66,25 @@ typedef Uint32 SDL_InitFlags;
...
+{
+    SDL_APP_CONTINUE,   /** Value that requests that the app continue from the main callbacks. If SDL_AppInit, SDL_AppEvent, or SDL_AppIterate returns this value, the program will continue to run. */
+    SDL_APP_SUCCESS,    /** Value that requests termination with success from the main callbacks. If SDL_AppInit, SDL_AppEvent, or SDL_AppIterate returns this value, the program will terminate and report success to the operating system. What that success looks like is platform-dependent. On Unix, for example, the process error code will be zero. */
+    SDL_APP_FAILURE     /** Value that requests termination with error from the main callbacks. If SDL_AppInit, SDL_AppEvent, or SDL_AppIterate returns this value, the program will terminate and report failure to the operating system. What that failure looks like is platform-dependent. On Unix, for example, the process error code will be non-zero. */
+} SDL_AppResult;
+
+typedef SDL_AppResult (SDLCALL *SDL_AppInit_func)(void **appstate, int argc, char *argv[]);
+typedef SDL_AppResult (SDLCALL *SDL_AppIterate_func)(void *appstate);
+typedef SDL_AppResult (SDLCALL *SDL_AppEvent_func)(void *appstate, const SDL_Event *event);
```
Comment (abouvier) [inline include/SDL3/SDL_init.h:85]: Shouldn't `SDL_events.h` be included in `SDL_init.h` because of this line?

Comment (slouken) [reply include/SDL3/SDL_init.h:85]: Yep, fixed in https://github.com/libsdl-org/SDL/commit/d29276e625836be3691e2e2ef2fdde206d19bc4c, thanks!

**Pull request merged**